src/mavlink-router/mainloop.cpp

```diff
@@ -755,47 +755,62 @@ void Mainloop::_handle_pipe()
 {
     char cmd[1024];
     ssize_t num_read = read(_pipefd, cmd, sizeof(cmd) - 1);
+    char* buffer = cmd;
     if (num_read > 0) {
         cmd[num_read] = 0;
         log_debug("Pipe read %ld bytes: %s", num_read, cmd);
-        // Op UDP Name IP Port Eavesdropping
-        std::vector<std::string> a;
-        std::string command = cmd;
-        char *pch = strtok(cmd, " ");
-        while (pch != nullptr) {
-            a.emplace_back(pch);
-            pch = strtok(nullptr, " \n");
-        }
-
-        if (a.size() == 2 && a[0] == "remove") {
-            _remove_dynamic_endpoint(a[1]);
-            return;
-        }
-
-        if (a.size() != 6 || a[0] != "add" || a[1] != "udp") {
-            return;
-        }
-        std::string name = a[2];
-        std::string address = a[3];
-        int port = atoi(a[4].c_str());
-        if (port <= 0) {
-            return;
-        }
-
-        auto pipecmd = _pipe_commands.find(name);
-        if (pipecmd != _pipe_commands.end() && pipecmd->second == command) {
-            auto ep = _dynamic_endpoints.find(name);
-            if (ep != _dynamic_endpoints.end()) {
-                ep->second->reset_expire_timer();
-                return;
-            }
-        }
 
-        std::unique_ptr<UdpEndpoint> udp{new UdpEndpoint()};
-        if (udp->open(address.c_str(), port, a[5]=="1") < 0) {
-            log_error("Could not open %s:%d", address.c_str(), port);
-            return;
+        // If more than one command separated by an end of
+        // line was written in the pipe, separate each command
+        char* current_new_line = strchr(buffer, '\n');
+        while (current_new_line != NULL) {
+          char command[1024] = {0};
+          strncpy(command, buffer, current_new_line - buffer);
```

Comment:
You can directly use `buffer` instead of `command`, if you do:
`*current_new_line = '\0';`

Reply:
I left the copy into command because it is used later in the code (in the if statement that leads to resetting the timer) but I want to update `buffer=current_new_line+1` before that given the `continue` in the other if statements. @bkueng is that okay?

Reply:
Yeah, that's fine. It's more important to avoid unnecessary memcpy's.